tests/shared/test_gcloud_client.py

```diff
@@ -226,20 +314,88 @@ def set_iam_policy(policy):
     grant_upload_access(EMAIL)
 
 
+def test_grant_bigquery_access(monkeypatch):
+    def set_iam_policy(resource, body):
+        assert resource == "CIDC-DFCI"
+        assert f"user:rando" in body["policy"]["bindings"][0]["members"]
+        assert f"user:{EMAIL}" in body["policy"]["bindings"][0]["members"]
+
+    def update_dataset(dataset, element):
+        assert element == ["access_entries"]
+        assert dataset.access_entries[0].entity_id == access_entry.entity_id
+        assert dataset.access_entries[1].entity_id == EMAIL
+
+    access_entry = bigquery.AccessEntry(
+        role="READER",
+        entity_type=EntityTypes.USER_BY_EMAIL,
+        entity_id="rando",
+    )
+
+    _mock_gcloud_bigquery_client(
+        monkeypatch,
+        [access_entry],
+        set_iam_policy,
+        update_dataset,
+    )
+
+    policy = {
+        "bindings": [{"role": GOOGLE_BIGQUERY_USER_ROLE, "members": ["user:rando"]}]
+    }
+    grant_bigquery_iam_access(policy, "CIDC-DFCI", [EMAIL])
+
+
 def test_revoke_upload_access(monkeypatch):
     def set_iam_policy(policy):
-        assert f"user:rando" in policy[GOOGLE_UPLOAD_ROLE]
-        assert f"user:{EMAIL}" not in policy[GOOGLE_UPLOAD_ROLE]
+        assert f"user:rando" in policy.bindings[0]["members"]
+        assert f"user:{EMAIL}" not in policy[1]["members"]
```

Comment:
these lines don't match, I think missing a `.bindings`

Reply:
actually, probably better for it to be
```suggestion
        assert any([f"user:rando" in b["members"] for b in policy.bindings])
        assert all([f"user:{EMAIL}" not in b["members"] for b in policy.bindings])
```